File: backend/src/Workora.Application/Features/Shifts/DTOs/ShiftDto.cs
using Workora.Shared.Responses;

namespace Workora.Application.Features.Shifts.DTOs;

/// <summary>
/// Data transfer object representing a shift schedule.
/// </summary>
public record ShiftDto(
    int Id,
    Guid Uuid,
    int CompanyId,
    int? BranchId,
    string Name,
    string Code,
    TimeOnly StartTime,
    TimeOnly EndTime,
    bool SpansMidnight,
    int GracePeriodMinutes,
    int BreakMinutes,
    string? Description,
    bool IsActive,
    DateTimeOffset CreatedAt);

File: frontend/workora-ui/src/app/data/dtos/shift.dto.ts
export interface ShiftDto {
  id: number;
  uuid: string;
  companyId: number;
  branchId?: number | null;
  name: string;
  code: string;
  startTime: string;
  endTime: string;
  spansMidnight: boolean;
  gracePeriodMinutes: number;
  breakMinutes: number;
  description?: string | null;
  isActive: boolean;
  createdAt: string;
}

export interface ShiftAssignmentDto {
  id: number;
  employeeId: number;
  shiftId: number;
  shiftName: string;
  startTime: string;
  endTime: string;
  effectiveFrom: string;
  effectiveTo?: string | null;
  isActive: boolean;
}

export interface CreateShiftRequestDto {
  companyId: number;
  name: string;
  code: string;
  startTime: string;
  endTime: string;
  spansMidnight: boolean;
  gracePeriodMinutes: number;
  breakMinutes: number;
  branchId?: number | null;
  description?: string | null;
}

export interface UpdateShiftRequestDto {
  name: string;
  code: string;
  startTime: string;
  endTime: string;
  spansMidnight: boolean;
  gracePeriodMinutes: number;
  breakMinutes: number;
  branchId?: number | null;
  description?: string | null;
}

export interface AssignShiftRequestDto {
  employeeId: number;
  shiftId: number;
  effectiveFrom: string;
  effectiveTo?: string | null;
}
